### PR TITLE
1.7 release note. [skip ci]

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by razdoburdin** [comment]
looks fine

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,161 @@ XGBoost Change Log
...
+
+### PySpark
+
+XGBoost 1.7 features initial support for PySpark integration. The new interface is adopted from the existing PySpark XGBoost interface developed by databricks with additional features like `QuantileDMatrix` and rapidsai plugin (GPU pipeline) support. The new Spark XGBoost Python estimators not only benefits from PySpark ml facilities for powerful distributed computing but also enjoys the rest of the Python ecosystem. Users can define a custom objective, callbacks, and metrics in Python and use them with this interface on distributed clusters. The support is labeled as experimental with more features to come in future releases. For a brief introduction please visit the tutorial on XGBoost's [document page](https://xgboost.readthedocs.io/en/latest/tutorials/spark_estimator.html). (#8355, #8344, #8335, #8284, #8271, #8283, #8250, #8231, #8219, #8245, #8217, #8200, #8173, #8172, #8145, #8117, #8131, #8088, #8082, #8085, #8066, #8068, #8067, #8020, #8385)
```
Comment (WeichenXu123) [inline NEWS.md:12]: Q: When can we remove the experimental label ?

**Comment by WeichenXu123** [comment]
LGTM

File: NEWS.md (hunk 1 above)
Comment (trivialfis) [reply NEWS.md:12]: @WeichenXu123 I think we will have at least one release for that. I will walk through the open issues for pyspark later on and revise a roadmap for improving it after 1.7.

Comment (trivialfis) [reply NEWS.md:12]: @WeichenXu123 I started a new github project on pyspark https://github.com/dmlc/xgboost/projects/5 .

Comment (trivialfis) [inline NEWS.md:12]:
```suggestion
XGBoost 1.7 features initial support for PySpark integration. The new interface is adapted from the existing PySpark XGBoost interface developed by databricks with additional features like `QuantileDMatrix` and the rapidsai plugin (GPU pipeline) support. The new Spark XGBoost Python estimators not only benefit from PySpark ml facilities for powerful distributed computing but also enjoy the rest of the Python ecosystem. Users can define a custom objective, callbacks, and metrics in Python and use them with this interface on distributed clusters. The support is labeled as experimental with more features to come in future releases. For a brief introduction please visit the tutorial on XGBoost's [document page](https://xgboost.readthedocs.io/en/latest/tutorials/spark_estimator.html). (#8355, #8344, #8335, #8284, #8271, #8283, #8250, #8231, #8219, #8245, #8217, #8200, #8173, #8172, #8145, #8117, #8131, #8088, #8082, #8085, #8066, #8068, #8067, #8020, #8385)
```

File: NEWS.md [hunk 2]
```diff
@@ -3,6 +3,161 @@ XGBoost Change Log
...
+
+Due to its initial support status, the new interface has some limitations including categorical features and multi-output models are not yet supported.
+
+### Development of categorical data support
+More progress on the experimental support for categorical features. In 1.7, XGBoost can handle missing values in categorical features and features an additional parameter `max_cat_threshold`, which limits the number of categories that can be used in the split evaluation. The parameter is enabled when partitioning algorithm is used and helps prevent over-fitting. Also, the sklearn interface can now accept the `feature_types` parameter to use data types other than dataframe for categorical features. (#8280, #7821, #8285, #8080, #7948, #7858, #7853, #8212, #7957, #7937, #7934)
```
Comment (trivialfis) [inline NEWS.md:17]:
```suggestion
More progress on the experimental support for categorical features. In 1.7, XGBoost can handle missing values in categorical features and features a new parameter `max_cat_threshold`, which limits the number of categories that can be used in the split evaluation. The parameter is enabled when the partitioning algorithm is used and helps prevent over-fitting. Also, the sklearn interface can now accept the `feature_types` parameter to use data types other than dataframe for categorical features. (#8280, #7821, #8285, #8080, #7948, #7858, #7853, #8212, #7957, #7937, #7934)
```

**Review by rongou — APPROVED** [review]

File: NEWS.md [hunk 3]
```diff
@@ -3,6 +3,161 @@ XGBoost Change Log
...
+
+* Java package now supports feature names and feature types for DMatrix in preparation for categorical feature support. (#7966)
+* Models trained by the JVM packages can now be safely used with other language bindings. (#7896, #7907)
+* Users can specify the model format when saving models with a stream. (#7940, #7955)
+* The default value for training parameters is now sourced from XGBoost directly, which helps JVM packages be consistent with other packages. (#7938)
+* Set the correct objective if the user doesn't explicitly set it (#7781)
+* Auto-detection of MUSL is replaced by system properties (#7921)
+* Improved error message for launching tracker. (#7952, #7968)
+* Fix a race condition in parameter configuration. (#8025)
+* [Breaking] ` timeoutRequestWorkers` is not removed. With the support for barrier mode, this parameter is no longer needed. (#7839)
```
Comment (hcho3) [inline NEWS.md:130]:
```suggestion
* [Breaking] ` timeoutRequestWorkers` is now removed. With the support for barrier mode, this parameter is no longer needed. (#7839)
```

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 4]
```diff
@@ -3,6 +3,161 @@ XGBoost Change Log
...
+With the growing adaption of the new internet protocol, XGBoost joined the club. In the latest release, the Dask interface can be used on IPv6 clusters, see XGBoost's Dask tutorial for details. (#8225, #8234)
+
+### Optimizations
+We have new optimizations for both the `hist` and `gpu_hist` tree methods to make XGBoost's training even more efficient.
+
+* Hist
+Hist now supports optional by-column histogram build, which is automatically configured based on various conditions of input data. This helps the XGBoost CPU hist algorithm to scale better with different shapes of training datasets. (#8233, #8259). Also, the build histogram kernel now can better utilize CPU registers (#8218)
+
+* GPU Hist
+GPU hist now supports batched node build, which reduces kernel latency and increases throughput. The improvement is particularly significant when growing deep trees with the default ``depthwise`` policy. (#7919, #8073, #8051, #8118, #7867, #7964, #8026)
```
Comment (RAMitchell) [inline NEWS.md:48]:
```suggestion
GPU hist performance is significantly improved for wide datasets. GPU hist now supports batched node build, which reduces kernel latency and increases throughput. The improvement is particularly significant when growing deep trees with the default ``depthwise`` policy. (#7919, #8073, #8051, #8118, #7867, #7964, #8026)
```

**Review by RAMitchell — APPROVED** [review]

**Review by hcho3 — APPROVED** [review]

**Pull request merged**